subprojects/core/src/main/java/org/gradle/api/internal/project/BuildOperationCrossProjectConfigurator.java

```diff
@@ -130,11 +93,27 @@ public void assertCrossProjectConfigurationAllowed(String methodName, Project ta
         return new Action<T>() {
             @Override
             public void execute(T t) {
-                nestedActionExecutionCount.set(nestedActionExecutionCount.get() + 1);
+                allowExection.set(false);
                 try {
                     action.execute(t);
                 } finally {
-                    nestedActionExecutionCount.set(nestedActionExecutionCount.get() - 1);
+                    allowExection.set(true);
```

Comment:
what if I do:
```
withCrossProjectConfigurationDisabled() {
    callSomethingElse()
    unsafeUserCode()
}

callSomethingElse() {
    withCrossProjectConfigurationDisabled() {
    }
}
```
Won't setting `allowExecution` to true after the inner call allow execution of the unsafe user code in the outer block?  Shouldn't we do something similar to below where we set the flag back to what it was before the call?

Reply:
good catch